### PR TITLE
refactor: move to vite + ts + swc

### PR DESCRIPTION
**Pull request by SrIzan10**
yeah
please @jacoobes take a look at it

### CODE REVIEW

**Comment by jacoobes** [comment]
fix build and ill review @SrIzan10

File: package.json [hunk 1]
```diff
@@ -17,24 +29,30 @@
...
+    "@typescript-eslint/eslint-plugin": "^5.61.0",
+    "@typescript-eslint/parser": "^5.61.0",
+    "@vitejs/plugin-react-swc": "^3.3.2",
+    "concurrently": "^8.2.0",
+    "electron": "^25.2.0",
+    "electron-builder": "^24.4.0",
+    "eslint": "^8.44.0",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-plugin-react-refresh": "^0.4.1",
+    "tsc-watch": "^6.0.4",
```
Comment (jacoobes) [inline package.json:46]: Why do you have ts-watch and vite? Doesn't vite have dev server 🤔

File: src/InitModal.tsx [hunk 2]
```diff
@@ -71,7 +71,7 @@ export default function InitModal() {
   };
 
   const [projectName, setProjectName] = React.useState('');
-  const handleProjectNameChange = (event) => {
+  const handleProjectNameChange = (event: any) => {
```
Comment (jacoobes) [inline src/InitModal.tsx]: event type should be `Event`

**Review by jacoobes — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (SrIzan10) [reply package.json:46]: tsc-watch is for electron file and vite is for website

File: src/InitModal.tsx (hunk 2 above)
Comment (SrIzan10) [reply src/InitModal.tsx]: didn't know what to put there, will fix

Comment (SrIzan10) [reply src/InitModal.tsx]: It ended up being `React.ChangeEvent<HTMLInputElement>`

**Comment by SrIzan10** [comment]
[Open in Gitpod](https://gitpod.io/#https://github.com/sern-handler/gui)
we need to also add that button to the readme file

**Review by jacoobes — APPROVED** [review]

**Comment by SrIzan10** [comment]
all right, looks like it works!
cheers! :tada:

**Pull request merged**